### PR TITLE
Feat/batch forest client data refresh

### PR DESCRIPTION
**Pull request by ianliuwk1019**
#248...

- Add CLIENT_API_TOKEN environment variable from OpenShift secret.
- Rename previous batch to '${NAME}-${ZONE}-work-flow-state-change-batch'.
- Add new CronJob for 'fc-client-data-refresh-batch'.
- Adjust name of the package.json batch run scripts.
- Install @nestjs/axios for use on NestJs Http.
- Minor refactor on constant.
- Add CLIENT API service configuration.
- Create new client-app-integration.module.ts and service for integration of ClIENT API.
- Configure ClientAppIntegrationModule with HttpModule instance using Axios.
- Add fetching logic with client-app-integration.service.ts for endpoint "findAllNonIndividuals".
- Use ClientAppIntegrationModule in forest-client.module.ts
- Adjust forest-client.service.ts and its controller away from ReadOnlyService to regular DataService to be able to persist.
- Implement fetch and refresh logic at forest-client service.
- Add second runBatch lgoc with -argv for new data refreshing job function.
---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-0.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-0.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-0.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by ianliuwk1019** [comment]
TODO: just notice, will need to add TEST/PROD environment: CLIENT_API_BASE_URL. Right now it is using default value.

**Comment by ianliuwk1019** [comment]
> TODO: just notice, will need to add TEST/PROD environment: CLIENT_API_BASE_URL. Right now it is using default value.

This has been added.

File: api/openshift.deploy.yml [hunk 1]
```diff
@@ -210,6 +210,16 @@ objects:
                     secretKeyRef:
                       key: data-encryption-key
                       name: ${NAME}-${ZONE}-${COMPONENT}
+                - name: CLIENT_API_BASE_URL
```
Comment (basilv) [inline api/openshift.deploy.yml:213]: I'm not sure this needed to be a secret - the URL isn't a secret, but it does need to be configured for each environment. Given that for the object storage we stored everything as secrets, this is being consistent so should be fine.

File: api/src/app/modules/app-config/configuration.ts [hunk 2]
```diff
@@ -32,4 +36,10 @@ export default registerAs('app', () => ({
   port: process.env.APP_PORT,
   globalPrefix: process.env.APP_PREFIX,
   title: process.env.APP_TITLE,
-}));
+
+  // Forest Client APP Integration
+  fcApiBaseUrl: process.env.CLIENT_API_BASE_URL,
+  fcApiApiToken: process.env.CLIENT_API_TOKEN,
+  fcApiReqTimeOut: process.env.CLIENT_API_REQ_TIMEOUT,
+  fcApiBatchSerchPageSize: process.env.CLIENT_API_BTH_PAGE_SIZE,
```
Comment (basilv) [inline api/src/app/modules/app-config/configuration.ts]: Spelling mistake -fcApiBatchSerchPageSize 'Serch' -> 'Search'.

File: api/src/app/modules/forest-client/forest-client.service.ts [hunk 3]
```diff
@@ -33,6 +42,44 @@ export class ForestClientService extends DataReadOnlyService<ForestClient, Repos
...
+        currentPage, 
+        this.configService.get("fcApiBatchSerchPageSize"),
+        ClientAppIntegrationService.SORT_BY_CLIENT_NUMBER
+      );
+
+      fetchedData.forEach(async (item) => {
+        let entity = await this.repository.findOne({ where: { id: item.id } })
+        if (!entity) {
+          entity = mapToEntity(item, new ForestClient());
+          entity.createUser = USER_SYSTEM;
```
Comment (basilv) [inline api/src/app/modules/forest-client/forest-client.service.ts:64]: Where do we set the create timestamp?

**Review by basilv — CHANGES_REQUESTED** [review]

File: api/openshift.deploy.yml (hunk 1 above)
Comment (ianliuwk1019) [reply api/openshift.deploy.yml:213]: Basil, your sense is very sharp. Yes, initially I was troubled with it and was thinking I might need to make this one not a secret, but after looking into other existing secret, I glued them together for consistency. Hope it is okay, we can change it if you change mind.

File: api/src/app/modules/app-config/configuration.ts (hunk 2 above)
Comment (ianliuwk1019) [reply api/src/app/modules/app-config/configuration.ts]: Thank you for finding it... I should double check again next time... This is fixed.

File: api/src/app/modules/forest-client/forest-client.service.ts (hunk 3 above)
Comment (ianliuwk1019) [reply api/src/app/modules/forest-client/forest-client.service.ts:64]: The db in fact has a default value for column `create_timestamp` but not on update_timestamp. So only `update` needs to set the time.

**Review by basilv — APPROVED** [review]

**Pull request merged**